centreon/www/class/centreonACL.class.php

```diff
@@ -226,9 +248,11 @@ private function setHostGroups()
         $query = "SELECT hg.hg_id, hg.hg_name, hg.hg_alias, arhr.acl_res_id "
             . "FROM hostgroup hg, acl_resources_hg_relations arhr "
             . "WHERE hg.hg_id = arhr.hg_hg_id "
-            . "AND hg.hg_activate = '1' "
-            . "AND arhr.acl_res_id IN (" . $this->getResourceGroupsString() . ") "
-            . "ORDER BY hg.hg_name ASC ";
+            . "AND hg.hg_activate = '1' ";
+        if ($this->hasAccessToAllHostGroups === false) {
+            $query.= "AND arhr.acl_res_id IN (" . $this->getResourceGroupsString() . ") ";
+        }
+        $query.= "ORDER BY hg.hg_name ASC ";
```

Comment:
even if you enhance the existing request could you use a prepare statement instead ?